lib/authn/util/poll.ts

```diff
@@ -77,10 +78,23 @@ export function getPollFn(sdk, res: AuthnTransactionState, ref) {
       }
 
       var href = pollLink.href + toQueryString(opts);
-      return post(sdk, href, getStateToken(res), {
+      const postPromise = post(sdk, href, getStateToken(res), {
         saveAuthnState: false,
         withCredentials: true
       });
+  
+      if (isIOS()) {
+        return Promise.race([
+          postPromise,
+          new Promise((_, reject) => {
```

Comment:
I'm not a fan of throwing an error to catch it later. I think this logic could be adjusted to fire the request, and a set a timer to fire the request again in 5s if the first request hasn't fulfilled. Something like:

```js
let didFulfill = false;
return Promise.race([
  postPromise.then(resp => { didFulFill = true; return resp }),
  delayFn(() => { if (!didFulfill) { return post(...); }
]);
```

Reply:
I think the retry logic should be handled same way as it's done in catch block for general network error (`Load failed` error) - with calling `delayNextPoll` and `recursivePoll` instead of just `post()` (to handle background situations)
https://github.com/okta/okta-auth-js/blob/3b91c02720979649ba3c1edd4cbab60bc4b6a4de/lib/authn/util/poll.ts#L188-L189